terraform/service/main.tf

```diff
@@ -0,0 +1,130 @@
+# Create a service running on fargate with a task definition and service definition
+terraform {
+  backend "s3" {
+    # Backend is selected using terraform init -backend-config=path/to/backend-<env>.tfbackend
+    # bucket         = "sdp-dev-tf-state"
+    # key            = "sdp-dev-ecs-example-service/terraform.tfstate"
+    # region         = "eu-west-2"
+    # dynamodb_table = "terraform-state-lock"
+  }
+
+}
+
+# Create CloudWatch Log Groups beforehand
+resource "aws_cloudwatch_log_group" "frontend_logs" {
+  name              = "/ecs/ecs-service-${var.service_subdomain}-frontend"
+  retention_in_days = var.log_retention_days
+}
+
+resource "aws_cloudwatch_log_group" "backend_logs" {
+  name              = "/ecs/ecs-service-${var.service_subdomain}-backend"
+  retention_in_days = var.log_retention_days
+}
+
+resource "aws_ecs_task_definition" "ecs_service_definition" {
+  family = "ecs-service-${var.service_subdomain}-application"
+  container_definitions = jsonencode([
+    {
+      # Frontend Container
+      name      = "${var.service_subdomain}-task-application"
+      image     = "${var.aws_account_id}.dkr.ecr.${var.region}.amazonaws.com/${var.frontend_ecr_repo}:${var.container_ver}"
+      cpu       = 512
```

Comment:
Should be variables.  Are you using the bioiler plate terraform from other projects? I don't think we have these hard coded.

Reply:
https://github.com/ONS-Innovation/sdp-ecs-fargate-service/blob/a0b232778a7e82fd461f9c72a4944ceec5ee3343/terraform/service/main.tf#L23-L25

Copied the name and image from here.

CPU I will change

Reply:
Added as variable + see message above.

Reply:
The setting of cpu that you linked to in sdp-ecs-fargate-service is setting the container allocation of cpu as shared (0), [further down](https://github.com/ONS-Innovation/sdp-ecs-fargate-service/blob/a0b232778a7e82fd461f9c72a4944ceec5ee3343/terraform/service/main.tf#L72) the task level CPU is defined using a variable.  This effectively allows us to have multiple containers running in a task and equally sharing CPU (we don't typically do this though).

Reply:
Ah okay, how should I do it? I've had it working like this (front and back containers in same task) with nginx and ALB.

I guess I presumed that, as the backend is so small and has only one endpoint that's used on the front, we wouldn't need individual tasks.

I can change though no worries.